pkg/nsx/services/nsxserviceaccount/cluster.go

```diff
@@ -0,0 +1,289 @@
+			IsProtected: &isProtectedTrue,
+			Name:        &normalizedClusterName,
+			NodeId:      &normalizedClusterName,
+			Role:        nil,
+			RolesForPaths: []mpmodel.RolesForPath{{
+				Path: &readerPath,
+				Roles: []mpmodel.Role{{
+					Role: &readerRole,
+				}},
+			}, {
+				Path: &vpcPath,
+				Roles: []mpmodel.Role{{
+					Role: &vpcRole,
+				}},
+			}},
+			CertificatePem: &cert,
+			Tags:           common.ConvertTagsToMPTags(s.buildBasicTags(obj)),
+		})
+		if err != nil {
+			return err
+		}
+		s.PrincipalIdentityStore.Add(pi)
+	}
+
+	// create ClusterControlPlane
+	clusterId := ""
+	if ccpObj := s.ClusterControlPlaneStore.GetByKey(normalizedClusterName); ccpObj == nil {
+		ccp, err := s.NSXClient.ClusterControlPlanesClient.Update(siteId, enforcementpointId, normalizedClusterName, model.ClusterControlPlane{
+			Revision:     &revision1,
+			ResourceType: &antreaClusterResourceType,
+			Certificate:  &cert,
+			VhcPath:      &vpcPath,
+			Tags:         s.buildBasicTags(obj),
+		})
+		if err != nil {
+			return err
+		}
+		s.ClusterControlPlaneStore.Add(ccp)
+		clusterId = *ccp.NodeId
+	}
+
+	// create Secret
+	secretName := obj.Name + SecretSuffix
+	secretNamespace := obj.Namespace
+	if err := s.Client.Create(ctx, &v1.Secret{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      secretName,
+			Namespace: secretNamespace,
+			// TODO: Add labels/annotations
+			Labels:      nil,
+			Annotations: nil,
+			OwnerReferences: []metav1.OwnerReference{{
+				APIVersion:         obj.APIVersion,
+				Kind:               obj.Kind,
+				Name:               obj.Name,
+				UID:                obj.UID,
+				Controller:         nil,
+				BlockOwnerDeletion: nil,
+			}},
+			Finalizers: nil,
+		},
+		Immutable: nil,
+		// TODO: Add NSX CA
+		Data: map[string][]byte{SecretCertName: []byte(cert), SecretKeyName: []byte(key)},
+		Type: "",
+	}); err != nil {
+		return err
+	}
+
+	// update NSXServiceAccountStatus
+	obj.Status.Phase = v1alpha1.NSXServiceAccountPhaseRealized
+	obj.Status.Reason = "Success."
+	obj.Status.NSXManagers = s.NSXConfig.NsxApiManagers
+	obj.Status.ClusterID = clusterId
+	obj.Status.ClusterName = clusterName
+	obj.Status.Secrets = []v1alpha1.NSXSecret{{
+		Name:      secretName,
+		Namespace: secretNamespace,
+	}}
+	obj.Status.VPCPath = vpcPath
+	// TODO: Add proxy
+	return s.Client.Status().Update(ctx, obj)
+}
+
+func (s *NSXServiceAccountService) DeleteNSXServiceAccount(ctx context.Context, namespacedName types.NamespacedName) error {
+	clusterName := s.getClusterName(namespacedName.Namespace, namespacedName.Name)
+	normalizedClusterName := util.NormalizeId(clusterName)
+	// delete Secret
+	secretName := namespacedName.Name + SecretSuffix
+	secretNamespace := namespacedName.Namespace
+	if err := s.Client.Delete(ctx, &v1.Secret{ObjectMeta: metav1.ObjectMeta{Name: secretName, Namespace: secretNamespace}}); err != nil && !errors.IsNotFound(err) {
+		log.Error(err, "Failed to delete", "secret", secretName, "namespace", secretNamespace)
```

Comment:
some log starts with lowercase, some with uppercase, could you ensure be consistent?

Reply:
Fixed.
All logs start with lowercase except proper noun.